### PR TITLE
FIX Several small fixes

### PR DESCRIPTION
**Pull request by ArturoAmorQ**
1. At the end of the [First look at the dataset notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/01_tabular_data_exploration.html), under the heading "we made important observations" we mention that linear models can only capture linear interactions but there is no prior reference to linear models.

2. The word "complex" at the end of the [Effect of the sample size in cross-validation notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/cross_validation_learning_curve.html) may be misleading. Here we don’t mean complexity as in the depth of a decision tree or the degree of a polynomial (which indeed would lead to overfitting), but rather that a more "expressive" model should be used.

3. In the [Solution for Exercise M3.02](https://inria.github.io/scikit-learn-mooc/python_scripts/parameter_tuning_sol_03.html) the wording refers to the "accuracy of the model" but the model is a `KNeighborsRegressor`.

4. The `value_counts` function in the [Classification metrics notebook](https://inria.github.io/scikit-learn-mooc/python_scripts/metrics_classification.html) is raising a warning.

This PR proposes some fixes to address those issues.

### CODE REVIEW

File: python_scripts/01_tabular_data_exploration.py [hunk 1]
```diff
@@ -364,5 +364,5 @@
 #   and evaluating your machine learning model;
 # * having redundant (or highly correlated) columns can be a problem for some
 #   machine learning algorithms;
-# * contrary to decision tree, linear models can only capture linear
-#   interactions, so be aware of non-linear relationships in your data.
+# * decision trees create rules at constant values of the features, such as
+#   vertical and horizontal lines in a 2-dimensional feature space.
```
Comment (ogrisel) [inline python_scripts/01_tabular_data_exploration.py:368]:
```suggestion
# * decision trees define prediction rules by comparing feature values to thresholds,
#   one feature after another. The resulting decision boundary of a decision tree
#   is therefore composed of vertical and horizontal line segments in a 2-dimensional
#   feature space.
```

File: python_scripts/01_tabular_data_exploration.py [hunk 2]
```diff
@@ -364,5 +364,5 @@
 #   and evaluating your machine learning model;
```
Comment (ogrisel) [inline python_scripts/01_tabular_data_exploration.py:364]: => "you may need to be careful when interpreting the values of performance metrics: for instance, the accuracy of a baseline prediction rule that constantly predicts the majority class can be much larger than 50%."

File: python_scripts/01_tabular_data_exploration.py [hunk 3]
```diff
@@ -364,5 +364,5 @@
...
 # * having redundant (or highly correlated) columns can be a problem for some
 #   machine learning algorithms;
```
Comment (ogrisel) [inline python_scripts/01_tabular_data_exploration.py:366]:
```suggestion
# * columns can be redundant (or highly correlated). This is not necessary a problem
#   for machine learning models but it can make it challenging to assess the
#   relative importance of the input variables in their ability to make accurate
#   predictions.    
```

**Review by ogrisel — COMMENTED** [review]
Thanks, here are some further improvements:

File: python_scripts/01_tabular_data_exploration.py (hunk 1 above)
Comment (ArturoAmorQ) [reply python_scripts/01_tabular_data_exploration.py:369]: Thanks for the review @ogrisel. I addressed your other comments, but this wording feels a bit too complex for an introductory notebook.

File: python_scripts/cross_validation_learning_curve.py [hunk 4]
```diff
@@ -104,7 +104,7 @@
 #
 # If we achieve a plateau and adding new samples in the training set does not
 # reduce the testing error, we might have reached the Bayes error rate using the
-# available model. Using a more complex model might be the only possibility to
+# available model. Using a more expressive model might be the only possibility to
 # reduce the testing error further.
```
Comment (ArturoAmorQ) [inline python_scripts/cross_validation_learning_curve.py:108]: @ogrisel If a model reached Bayes error then isn’t that error is irreducible meaning if we even select a more expressive model that won’t improve our test error?

Comment (ArturoAmorQ) [reply python_scripts/cross_validation_learning_curve.py]: Wording fixed in [bcc45f3](https://github.com/INRIA/scikit-learn-mooc/pull/780/commits/bcc45f3adb6df161780ab042d9b222e2d7259066).

File: python_scripts/01_tabular_data_exploration.py (hunk 1 above)
Comment (ogrisel) [reply python_scripts/01_tabular_data_exploration.py:369]: > Thanks for the review @ogrisel. I addressed your other comments, but this wording feels a bit too complex for an introductory notebook.

Personally, I find the previous wording more confusing: "create rules at constant values of the features" does not mean anything. What does a "rule at a value" mean?

Maybe the following is both simpler and actually meaningful:

```suggestion
# * decision trees define prediction rules that successively compare each feature to
#   a threshold value. The resulting decision boundary of a decision tree is
#   therefore composed of vertical and horizontal line segments in a 2-dimensional
#   feature space.
```

File: python_scripts/cross_validation_learning_curve.py [hunk 5]
```diff
@@ -102,10 +102,11 @@
...
 #
-# If we achieve a plateau and adding new samples in the training set does not
-# reduce the testing error, we might have reached the Bayes error rate using the
-# available model. Using a more complex model might be the only possibility to
-# reduce the testing error further.
+# If the testing error plateaus despite adding more training samples, it's
+# possible that the model has achieved its optimal performance. In this case,
+# using a more expressive model might help reduce the error further. Otherwise,
+# the error may have reached the Bayes error rate, the theoretical minimum error
+# due to inherent uncertainty in the data.
```
Comment (ogrisel) [inline python_scripts/cross_validation_learning_curve.py:109]: It might be helpful to explain why this is expected:

```suggestion
# due to inherent uncertainty not resolved by the available data. This minimum error is
# non-zero whenever some of the variation of the target variable `y` depends on
# external factors not fully observed in the features available in `X`, which is
# almost always the case in practice.
```

**Review by ogrisel — APPROVED** [review]
Some more feedback. LGTM otherwise.

**Comment by ogrisel** [comment]
@ArturoAmorQ I finally answered your feedback in https://github.com/INRIA/scikit-learn-mooc/pull/780#discussion_r1853981184.

File: python_scripts/01_tabular_data_exploration.py [hunk 6]
```diff
@@ -360,9 +360,12 @@
...
-#   interactions, so be aware of non-linear relationships in your data.
+#   target category than another), you may need to be careful when interpreting
+#   the values of performance metrics;
+# * columns can be redundant (or highly correlated), which is not necessarily a
+#   problem, but may require special treatment as we will cover in future
+#   notebooks;
+# * decision trees create prediction rules by comparing each feature to a
+# threshold value, resulting in decision boundaries that are always parallel to
+# the axes. In 2D, this means the boundaries are vertical or horizontal line
+# segments at the feature threshold values.
```
Comment (ArturoAmorQ) [inline python_scripts/01_tabular_data_exploration.py:371]:
```suggestion
# * decision trees create prediction rules by comparing each feature to a
#   threshold value, resulting in decision boundaries that are always parallel
#   to the axes. In 2D, this means the boundaries are vertical or horizontal
#   line segments at the feature threshold values.
```
